internal/manifest/ready_check.go

```diff
@@ -2,227 +2,67 @@ package manifest
+	deploymentState := getDeploymentState(clnt, resources)
+	return declarativev2.StateInfo{State: deploymentState}, nil
 }
 
-func HandleState(manifest *v1beta2.Manifest, moduleCR *unstructured.Unstructured) (declarativev2.StateInfo, error) {
-	stateChecks, customStateFound, err := parseStateChecks(manifest)
-	if err != nil {
-		return declarativev2.StateInfo{State: shared.StateError}, fmt.Errorf(
-			"could not get state from module CR %s to determine readiness: %w",
-			moduleCR.GetName(), err,
-		)
-	}
-	typedState, stateExists, err := mappingState(stateChecks, moduleCR, customStateFound)
-	if err != nil {
-		// Only happens for kyma module CR
-		if errors.Is(err, ErrNotSupportedState) {
-			return declarativev2.StateInfo{
-				State: shared.StateWarning,
-				Info:  ErrNotSupportedState.Error(),
-			}, nil
-		}
-		return declarativev2.StateInfo{State: shared.StateError}, fmt.Errorf(
-			"could not get state from module CR %s to determine readiness: %w",
-			moduleCR.GetName(), err,
-		)
-	}
-	if !stateExists {
-		info := ModuleCRWithNoCustomCheckWarning
-		if customStateFound {
-			info = ModuleCRWithCustomCheckWarning
-		}
-		state := shared.StateProcessing
-		// If wait for certain period of time, state still not found, put manifest state into Warning
-		if manifest.CreationTimestamp.Add(ToWarningDuration).Before(time.Now()) {
-			state = shared.StateWarning
-		}
-
-		return declarativev2.StateInfo{State: state, Info: info}, nil
-	}
-
-	return declarativev2.StateInfo{State: typedState}, nil
-}
-
-func mappingState(stateChecks []*v1beta2.CustomStateCheck,
-	moduleCR *unstructured.Unstructured,
-	customStateFound bool,
-) (shared.State, bool, error) {
-	// make sure ready and error state exists, for other missing customized state, can be ignored.
-	if requiredStateMissing(stateChecks) {
-		return "", false, ErrRequiredStateMissing
-	}
-	stateResult := map[shared.State]bool{}
-	foundStateInCR := false
-	for _, stateCheck := range stateChecks {
-		stateFromCR, stateExists, err := unstructured.NestedString(moduleCR.Object,
-			strings.Split(stateCheck.JSONPath, ".")...)
-		if err != nil {
-			return "", false, fmt.Errorf("could not get state from module CR %s at path %s "+
-				"to determine readiness: %w", moduleCR.GetName(), stateCheck.JSONPath, err)
-		}
-		if !stateExists {
-			continue
-		}
-		if !customStateFound && !shared.State(stateFromCR).IsSupportedState() {
-			return "", false, ErrNotSupportedState
-		}
-		foundStateInCR = true
-		_, found := stateResult[stateCheck.MappedState]
-		if found {
-			stateResult[stateCheck.MappedState] = stateResult[stateCheck.MappedState] && stateFromCR == stateCheck.Value
-		} else {
-			stateResult[stateCheck.MappedState] = stateFromCR == stateCheck.Value
-		}
+func getDeploymentState(clt declarativev2.Client, resources []*resource.Info) shared.State {
+	deploy, found := findDeployment(clt, resources)
+	// Not every module operator use Deployment by default, e.g: StatefulSet also a valid approach
```

Comment:
Let's have a followup issue to support `StatefulSet` state check.

Reply:
https://github.com/kyma-project/lifecycle-manager/issues/1649